src/Nethermind/Nethermind.Network/P2P/ProtocolHandlers/SyncPeerProtocolHandlerBase.cs

```diff
@@ -183,7 +182,18 @@ public virtual Task<byte[][]> GetNodeData(IReadOnlyList<Keccak> hashes, Cancella
 
         public void SendNewTransaction(Transaction tx)
         {
-            SendMessage(new[] { tx });
+            if (tx.Hash != null && NotifiedTransactions.Set(tx.Hash))
```

Comment:
QQ: we haven't had this if `tx.Hash != null && NotifiedTransactions.Set(tx.Hash)` before. Is it fixing some bug, but not related to 4844?

Reply:
It is recently added feature https://github.com/NethermindEth/nethermind/pull/5449
It was changed for `SendNewTransactions`:
https://github.com/NethermindEth/nethermind/pull/5449/files#diff-e2801b05e4ee2fda5592da789a82f0e4c733c8b20c8688020cbb0de71e3db5e5
and `SendNewTransaction` was skipped. I just added it here too.
It is a protection from sending the same tx few times to one peer - we are caching sent transactions. Not directly related to 4844